ap.md

```diff
@@ -0,0 +1,228 @@
+{% endhighlight %}
+
+Inside our `createServer` callback function, we've replaced the old plain text response with a basic router that catches requests to '/leds/:index' and defaults all other requests to a separate function. Let's add those functions below the `console.log`:
+
+{% highlight javascript %}
+// respond to the request with our index.html page
+function showIndex (url, request, response) {
+  response.writeHead(200, {"Content-Type": "text/html"});
+  // use fs to read in index.html
+  fs.readFile(__dirname + '/index.html', function (err, content) {
+    if (err) {
+      throw err;
+    }
+
+    // serves the content of index.html read in by fs
+    response.end(content);
+  });
+}
+
+// toggle the led specified in the url and respond with its state
+function toggleLED (url, request, response) {
+  var indexRegex = /(\d)$/; // Regex to find the number at the end of the url
+  var result = indexRegex.exec(url); // capture the number, returns an array
+  var index = +result[1]; // grab the captured result from the array and make sure it's a Number
+
+  var led = tessel.led[index]; // use the index to refence the correct LED
+
+  // toggle the state of the led and call the callback after that's done
+  led.toggle(function (err) {
+    if (err) {
+      // log the error and send back a 500 (internal server error) response to the client
+      console.log(err);
+      response.writeHead(500, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({error: err}));
+    } else {
+      // the led was successfully toggled, so respond with the state of the toggled led using led.isOn
+      response.writeHead(200, {"Content-Type": "application/json"});
+      response.end(JSON.stringify({on: led.isOn}));
+    }
+  });
+}
+{% endhighlight %}
+
+In our `showIndex` function, we grab the content of `index.html` using the Node.js `fs` module and respond to the web browser request with that content. In our toggleLED function, we use a Regex to search the url for the index of the led we want to toggle and pass that index into the `tessel.led` array. Now that we have the led we want, we toggle it and use the callback to respond to the server request with the current state of the led using `isOn` property.
+
+One last step is to create a `.tesselignore` file in the project directory and add any non-JavaScript files we want to deploy with `index.js`. For this project, that means adding one line: `index.html`. 
+
+Finally, let's fire up our server again by running:
+
+`t2 run index.js`
+
+Just like before, once you see the "Server running at http://192.168.1.101:8080/" message in your terminal, you should be able to connect to that url in the web browser of the device connected to the TesselRouter network. After connecting, you should see the index.html view we built earlier and, after clicking/tapping the buttons, see either the blue or green LEDs on the Tessel should be toggled on or off. 
+
+[Show demo video here?]
+
```

Comment:
I've been thinking about that. The ideas I've come up with so far are more JavaScript specific than Tessel, like "try replacing XHR with WebSockets" or "try adding some controls to the web page for making the LEDs blink at various intervals". 

Maybe, like the tweet tutorial, the bonus can involve using a Tessel module. For example: "As a bonus, try adding some UI, like a slider, to the web page for controlling a servo module."